Compilation broken on MinGW

It seems like some of the last wasapi changes (looks like #164 but might have been before) broke compilation on mingw-w64 (not sure about other compilers on windows, might be broken as well).
f26ba07 still works for me on the current setup.
I am using gcc version 7.1, cmake and ninja without any additional options:

```
[26/48] Building C object CMakeFiles/libsoundio_shared.dir/src/wasapi.c.obj
FAILED: C:\PROGRA~1\mingw64\bin\gcc.exe  -Dlibsoundio_shared_EXPORTS -I../ -I. -I../test -I../src -g -Werror -pedantic   -std=c11 -fvisibili
ty=hidden -Wall -Werror=strict-prototypes -Werror=old-style-definition -Werror=missing-prototypes -D_REENTRANT -D_POSIX_C_SOURCE=200809L -Wn
o-missing-braces -MD -MT CMakeFiles/libsoundio_shared.dir/src/wasapi.c.obj -MF CMakeFiles\libsoundio_shared.dir\src\wasapi.c.obj.d -o CMakeF
iles/libsoundio_shared.dir/src/wasapi.c.obj   -c C:\Users\nyorain\build\libsoundio\src\wasapi.c
In file included from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/winbase.h:2380:0,
                 from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/windows.h:70,
                 from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/rpc.h:16,
                 from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/audioclient.h:7,
                 from C:\Users\nyorain\build\libsoundio\src\wasapi.c:17:
C:\Users\nyorain\build\libsoundio\src\wasapi.c: In function 'outstream_do_open':
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1383:64: error: incompatible type for argument 2 of 'osw->audio_client->lpVtbl->GetService'
     if (FAILED(hr = IAudioClient_GetService(osw->audio_client, IID_ISimpleAudioVolume,
                                                                ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1383:64: note: expected 'const IID * const {aka const struct _GUID * const}' but argument is
of type 'GUID {aka const struct _GUID}'
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1389:46: error: 'ISimpleAudioVolume {aka struct ISimpleAudioVolume}' has no member named 'Get
MasterVolume'
     if (FAILED(hr = osw->audio_volume_control->GetMasterVolume(&volume)))
                                              ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1389:65: error: 'volume' undeclared (first use in this function)
     if (FAILED(hr = osw->audio_volume_control->GetMasterVolume(&volume)))
                                                                 ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1389:65: note: each undeclared identifier is reported only once for each function it appears
in
C:\Users\nyorain\build\libsoundio\src\wasapi.c: In function 'outstream_set_volume_wasapi':
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1728:46: error: 'ISimpleAudioVolume {aka struct ISimpleAudioVolume}' has no member named 'Set
MasterVolume'
     if (FAILED(hr = osw->audio_volume_control->SetMasterVolume(&volume)))
                                              ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1727:13: error: variable 'hr' set but not used [-Werror=unused-but-set-variable]
     HRESULT hr;
             ^~
cc1.exe: all warnings being treated as errors
[26/48] Building C object CMakeFiles/latency.dir/src/wasapi.c.obj
FAILED: C:\PROGRA~1\mingw64\bin\gcc.exe   -I../ -I. -I../test -I../src -g -Werror -pedantic   -std=c11 -fvisibility=hidden -Wall -Werror=str
ict-prototypes -Werror=old-style-definition -Werror=missing-prototypes -D_REENTRANT -D_POSIX_C_SOURCE=200809L -Wno-missing-braces -MD -MT CM
akeFiles/latency.dir/src/wasapi.c.obj -MF CMakeFiles\latency.dir\src\wasapi.c.obj.d -o CMakeFiles/latency.dir/src/wasapi.c.obj   -c C:\Users
\nyorain\build\libsoundio\src\wasapi.c
In file included from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/winbase.h:2380:0,
                 from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/windows.h:70,
                 from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/rpc.h:16,
                 from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/audioclient.h:7,
                 from C:\Users\nyorain\build\libsoundio\src\wasapi.c:17:
C:\Users\nyorain\build\libsoundio\src\wasapi.c: In function 'outstream_do_open':
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1383:64: error: incompatible type for argument 2 of 'osw->audio_client->lpVtbl->GetService'
     if (FAILED(hr = IAudioClient_GetService(osw->audio_client, IID_ISimpleAudioVolume,
                                                                ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1383:64: note: expected 'const IID * const {aka const struct _GUID * const}' but argument is
of type 'GUID {aka const struct _GUID}'
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1389:46: error: 'ISimpleAudioVolume {aka struct ISimpleAudioVolume}' has no member named 'Get
MasterVolume'
     if (FAILED(hr = osw->audio_volume_control->GetMasterVolume(&volume)))
                                              ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1389:65: error: 'volume' undeclared (first use in this function)
     if (FAILED(hr = osw->audio_volume_control->GetMasterVolume(&volume)))
                                                                 ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1389:65: note: each undeclared identifier is reported only once for each function it appears
in
C:\Users\nyorain\build\libsoundio\src\wasapi.c: In function 'outstream_set_volume_wasapi':
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1728:46: error: 'ISimpleAudioVolume {aka struct ISimpleAudioVolume}' has no member named 'Set
MasterVolume'
     if (FAILED(hr = osw->audio_volume_control->SetMasterVolume(&volume)))
                                              ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1727:13: error: variable 'hr' set but not used [-Werror=unused-but-set-variable]
     HRESULT hr;
             ^~
cc1.exe: all warnings being treated as errors
[26/48] Building C object CMakeFiles/libsoundio_static.dir/src/wasapi.c.obj
FAILED: C:\PROGRA~1\mingw64\bin\gcc.exe   -I../ -I. -I../test -I../src -g -Werror -pedantic   -std=c11 -fvisibility=hidden -Wall -Werror=str
ict-prototypes -Werror=old-style-definition -Werror=missing-prototypes -D_REENTRANT -D_POSIX_C_SOURCE=200809L -Wno-missing-braces -MD -MT CM
akeFiles/libsoundio_static.dir/src/wasapi.c.obj -MF CMakeFiles\libsoundio_static.dir\src\wasapi.c.obj.d -o CMakeFiles/libsoundio_static.dir/
src/wasapi.c.obj   -c C:\Users\nyorain\build\libsoundio\src\wasapi.c
In file included from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/winbase.h:2380:0,
                 from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/windows.h:70,
                 from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/rpc.h:16,
                 from C:/PROGRA~1/mingw64/x86_64-w64-mingw32/include/audioclient.h:7,
                 from C:\Users\nyorain\build\libsoundio\src\wasapi.c:17:
C:\Users\nyorain\build\libsoundio\src\wasapi.c: In function 'outstream_do_open':
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1383:64: error: incompatible type for argument 2 of 'osw->audio_client->lpVtbl->GetService'
     if (FAILED(hr = IAudioClient_GetService(osw->audio_client, IID_ISimpleAudioVolume,
                                                                ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1383:64: note: expected 'const IID * const {aka const struct _GUID * const}' but argument is
of type 'GUID {aka const struct _GUID}'
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1389:46: error: 'ISimpleAudioVolume {aka struct ISimpleAudioVolume}' has no member named 'Get
MasterVolume'
     if (FAILED(hr = osw->audio_volume_control->GetMasterVolume(&volume)))
                                              ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1389:65: error: 'volume' undeclared (first use in this function)
     if (FAILED(hr = osw->audio_volume_control->GetMasterVolume(&volume)))
                                                                 ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1389:65: note: each undeclared identifier is reported only once for each function it appears
in
C:\Users\nyorain\build\libsoundio\src\wasapi.c: In function 'outstream_set_volume_wasapi':
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1728:46: error: 'ISimpleAudioVolume {aka struct ISimpleAudioVolume}' has no member named 'Set
MasterVolume'
     if (FAILED(hr = osw->audio_volume_control->SetMasterVolume(&volume)))
                                              ^
C:\Users\nyorain\build\libsoundio\src\wasapi.c:1727:13: error: variable 'hr' set but not used [-Werror=unused-but-set-variable]
     HRESULT hr;
             ^~
cc1.exe: all warnings being treated as errors
[26/48] Building C object CMakeFiles/unit_tests.dir/src/dummy.c.obj
ninja: build stopped: subcommand failed.         
```
I have sadly no real clue about this winapi guid, iid, whatever stuff so will probably not be able to fix it.
CI (travis and appveyor) would probably be nice to detect compilation issues like this, was there any (fixable) reason for removing travis a while ago?

--- a/src/wasapi.c
+++ b/src/wasapi.c
@@ -1386,7 +1386,7 @@ static int outstream_do_open(struct SoundIoPrivate *si, struct SoundIoOutStreamP
         return SoundIoErrorOpeningDevice;
     }
 
-    if (FAILED(hr = osw->audio_volume_control->GetMasterVolume(&volume)))
+    if (FAILED(hr = osw->audio_volume_control->lpVtbl->GetMasterVolume(osw->audio_volume_control, &outstream->volume)))
     {
         return SoundIoErrorOpeningDevice;
     }
@@ -1725,7 +1725,7 @@ static int outstream_set_volume_wasapi(struct SoundIoPrivate *si, struct SoundIo
     struct SoundIoOutStreamWasapi *osw = &os->backend_data.wasapi;
 
     HRESULT hr;
-    if (FAILED(hr = osw->audio_volume_control->SetMasterVolume(&volume)))
+    if (FAILED(hr = osw->audio_volume_control->lpVtbl->SetMasterVolume(osw->audio_volume_control, volume, NULL)))
     {
         return SoundIoErrorIncompatibleDevice;
     }

